src/common/buf.c

```diff
@@ -203,3 +203,33 @@ buf_move(struct buf *dst, struct buf *src)
 	*dst = *src;
 	*src = BUF_INIT;
 }
+
+struct buf
+buf_from_file(const char *filename)
+{
+	struct buf buf = BUF_INIT;
+	FILE *stream = fopen(filename, "r");
+	if (!stream) {
+		return buf;
+	}
+
+	if (fseek(stream, 0, SEEK_END) == -1) {
+		fclose(stream);
+		return buf;
+	}
+	long size = ftell(stream);
+	if (size == -1) {
+		fclose(stream);
+		return buf;
+	}
+	rewind(stream);
+
+	buf_expand(&buf, size + 1);
```

Comment:
Why the `+ 1` here?

Other than that looks fine but lets give @johanmalm a chance to review.

Reply:
Yeah, doing `ftell()` on a FILE generated like this `printf '%b' "123" > test.txt` returns 3, so need to allow space for null byte.